src/visitor.rs

```diff
@@ -161,6 +154,13 @@ impl<'a> FmtVisitor<'a> {
             visit::FnKind::Closure => None,
         };
 
+        if let Some(ref single_line_fn) = self.rewrite_single_line_fn(&rewrite, &b) {
```

Comment:
It would require a ton of refactoring, since `visit_block` goes down the rabbit hole and offloads a lot of work to `visit_stmt`, `visit_item`, `visit_mac`, etc. A lot of the visitor code would need to be stripped into rewrite functions.